### ISSUE TITLE
GOG Galaxy 2.0.80 Beta Breaks Game installation (Error E2)

### ISSUE THREAD

**Issue by Ryonez** [moraroy/NonSteamLaunchers-On-Steam-Deck#515]
This is more a notice, you can work around it but users need to be aware not to update the client.

There's a link on GOG's forms with more info here, [GOG Galaxy 2.0.80 Beta](https://www.gog.com/forum/general_beta_gog_galaxy_2.0/gog_galaxy_2080_beta_breaks_steam_deck_linux_compatibility), but the jist is the new 2.0.80 Beta client results with games having E2 errors when installing. No game will work with the client atm.
Rolling back to version 2.0.77.22 of the client does work, but you need to make sure not to update for now.
For myself, I replaced the GOG client link in the plugin to one of this older version to install it.

**Comment by moraroy** [comment]
@Ryonez does steams Proton 9.0.3 help?

**Comment by HazedNapkin** [comment]
I was having this issue and changing to proton 9.0.3 worked for me, thanks

**Comment by LaiKash** [comment]
Tried with 9.0.4 and didn't work. I'll try uninstalling too

**Comment by razorblade446** [comment]
Followed @Ryonez advice and solved the issue temporarily. Maybe we need some way of providing the stable version for GoG (Because of it providing a BETA)

**Comment by moraroy** [comment]
@razorblade446 @Ryonez @LaiKash @HazedNapkin  I've been using Proton GE 9-10 actually guys, seems to work very well, not sure if this is still a problem with the "updated version of gog" yet though

**Comment by jamiehall123** [comment]
I ended up following a gog help article which worked for me. When it's in the final install stages (don't close gog galaxy), in the gog program folder rename the Dependencies-Temp folder to Dependancies, minute or two later it finishes the install successfully.

https://support.gog.com/hc/en-us/articles/360013537357-GOG-GALAXY-Installation-failed-E2-Update-failed-E2-Failed-to-finalize-the-update?product=gog

**Comment by moraroy** [comment]
@jamiehall123 amazing..... could you do a pull request?

**Comment by jamiehall123** [comment]
Sorry I should have clarified, the folder rename has to happen every time you install/update/verify a game in gog galaxy to avoid the E2 error.

I would class this as a workaround as it seems the Dependancies aren't installing, or being detected properly in this latest version.

Presuming renaming the folder forces the gog client into thinking the Dependancies have installed.

**Comment by moraroy** [comment]
@jamiehall123 well that's annoying lol I'll see keep an eye on it thank you for finding this! <3

**Comment by Ryonez** [comment]
I was having trouble were the folder rename wasn't helping on the problematic version.

**Comment by jamiehall123** [comment]
> I was having trouble were the folder rename wasn't helping on the problematic version.

Just checking, did you perform the rename while the client was running, and towards the end of the game installing/updating?

**Comment by Ryonez** [comment]
> > I was having trouble were the folder rename wasn't helping on the problematic version.
> 
> Just checking, did you perform the rename while the client was running, and towards the end of the game installing/updating?

Was after it failed and I closed the client and switched to desktop.
My goal was to have this all working while in gamemode, the only concreate way I could do this was reverting back to the prior working version, 2.0.77.22 in this case. The proton version at the time made no difference, I've not tried Proton GE 9-10 yet.

I'm guessing if the updated client is a problem, and if Proton GE 9-10 does work, then GOG changed some filesystem process somewhere, that the newer GE adds. I've no idea what atm though, I haven't touched it since I rolled the client version back.

**Comment by jbroberg** [comment]
If nothing has changed in the most recent release i think it's fair to say this still doesn't work out the box.

My fix was installing Proton GE 9-10 via protonup-qt manually and force running gog via this speific version after installing GoG via NSL.

Proton GE 9-10 wasn't an installed proton version option out of the box after plugin install - maybe it should be if it's currently required to install GoG games?

many thanks for the NSL work it's amazing.

**Comment by moraroy** [comment]
@jbroberg yeah I'm considering that, not sure if a new proton will fix it or should I wait for gog to fix this issue. I'll give it a little more time. In this new update alot of the UMU shortcuts for gog were broken so be sure to let NSL remake those shortcuts. Using proton ge 9-10 seems to still work

**Comment by rogerclotet** [comment]
I have the same issue, didn't try installing GOG Galaxy before yesterday, the client seems to work okay but I have the same install problems reported.
I tried renaming the Dependencies directory as a workaround at different times during the installation but it doesn't seem to make a difference.

**Comment by moraroy** [comment]
@rogerclotet what about GE Proton 9-10?

**Comment by rogerclotet** [comment]
Ok, you're right @moraroy, I only tried with the latest GE Proton (9-22). I just tried 9-10 and it works! Thank you!

**Comment by Karrq** [comment]
Can confirm, GE Proton 9-10 works :)

**Comment by michaelb2225** [comment]
> Can confirm, GE Proton 9-10 works :)

Dumb question - EVERY time I go to load GOG via game mode I have to manually change the compatibility to Proton 9-10. What am I missing here?

**Comment by moraroy** [comment]
@michaelb2225 not missing anything there is nothing in the script yet that sets it for gog yet, as usually it sets the latest ge proton.

**Comment by denny-dee** [comment]
Not sure if my points are relevant to this thread, but I'll add my 2 cents. It all happened sometime in November-December I think, didn't play much NSL games since then.
My GOG version turns out to be 2.0.80 (I don't remember signing up for beta).
I have issues for the past 2-3 months, where I can barely play any NSL games other than EGS.
I've had issues running Sniper Ghost Warrior 3 from GOG, until I reinstalled the GOG client (unsure if I started fresh or just uninstalled and reinstalled GOG). Also, I had to disable GOG cloud saves, as they lead to crash. Also, I believe I had some troubles with UMU links for some game (maybe this one), until I replaced those with pre-umu links (just copied from some older and working game, changed ID and file path).
I installed and played Just Cause 1 and 2 from GOG a couple of times, until the SteamOS and NSL update (unsure if it was December or November). After that, I had issue with cloud saves for these games, and disabling this feature didn't work for me, had to uninstall the games. Maybe some file permission issues prevented GOG from saving cloud saves locally?
Now I've installed Far Cry 2 from GOG, but the game does not start, probably because of the client version. 
Also, maybe NSL shouldn't force compatibility through launch options? If we force Proton through Steam UI, I'm not sure if it's used for the game now.

Also, I installed EA App back in December. I've played NFS for a few hours, it ran perfectly. However, now EA needs an update. It restarts with no issues, downloads the update, and fails to apply the update. I've tried fixing it by updating through NSL (both script and Decky plugin), but it crashes during the update. Again, file permissions issue maybe?

**Comment by Karrq** [comment]
I also noticed the proton version being set, but I believe it could be just the default for newly added shortcuts. Didn't really check if, after changing it manually, the version was changed back

**Comment by moraroy** [comment]
@denny-dee 1. when running the gog launcher and its games use Proton GE 9-10, it shouldn't crash anymore.   2. There was an error in the umu shortcuts back then, please get rid of all umu shortcuts and have NSL remake them, you should be good to go.  3. With umu, as far as I can tell, I don't think I'm able to get rid of the proton path inside of the command, that's why for umu shortcuts there is no force compatability being set and its inside of the launch options instead. Umu launcher seems to need that command. 4. Use the "repair EA App" inside of your library, when you installed the EA App, another shortcut was given to you called "repair ea  app" this will repair the app, when ea decides to boot loop.

**Comment by moraroy** [comment]
@Karrq yes I belive this happens on the desktop version only correct? I noticed this too, I may be able to fix.

**Comment by Count-Duckula-AU** [comment]
Thanks for the tip to use Proton GE 9-10 folks, seems to have fixed all my GOG issues :)

The GOG client updated a while back and errored while doing so. Since then the client would load but either get stuck on 'syncing' or crash at that point when launching a game. Installing a couple of new games would also partly work (shows them as installed but errored at the end of the install process, verify always failed).

Tried letting NSL reinstall the gog client but the gog installer would report a newer version was present and close.

**Comment by pivorsc** [comment]
Hello, just adding my 2 cents, did not play any GoG games for a while and today i run into very akward GoG client behaviour, couldnt run some games (E2), install or uninstall, tried fresh start, did not help aswell, entire client was not responsible at all.

Also one akward thing i noticed is that some of the standalone installed games through steam (gog standalone installers), have disappeared from my steam library and have appeared in gog client.. i do this when game dont work from gog client but is fine through adding non steam games to the library.

I dont know how this is even possible, blaming NSL is just my guess but i dont see any other way how this could happen.

**Comment by moraroy** [comment]
@pivorsc make sure all of your gog shortcuts and the launcher is using proton ge 9 -10

**Comment by pivorsc** [comment]
@moraroy i only got proton ge 11, 23 and 24, tried them all, managed to do fresh install, but then again, some games did not work and i couldnt uninstall anything

what is even more anoying is that, once again, the games i installed through "add non-steam game" yesterday, have just disappeared, this only affects gog games, i dont know what is going on

**Comment by moraroy** [comment]
@pivorsc you said you pressed  "Start Fresh" that literally wipes everything. You have no more games installed until you install them back from the launchers.

**Comment by moraroy** [comment]
@Ryonez @HazedNapkin @LaiKash @razorblade446 @jamiehall123 @jbroberg @rogerclotet @Karrq @denny-dee @Count-Duckula-AU @pivorsc 

wanted to tag everyone, so i believe i have fixed gog for NSL, got a new url, tested this and it seems you no longer need to use just proton ge 9-10 , i used the latest version proton GE 9-26  and it seemed to work for my games so im going to add in the fix to the Desktop Version (NonSteamLaunchers.desktop)

ive been doing alot of fixes for the desktop version in regards to the compat tool bugs, hopefully  those are better now

With this new gog launcher, you would need to use NSL and Uninstall GOG Galaxy and then Install it again, this will install the new version.  If you simply "re-install" it wont work. so uninstall  gog and then install  gog

so far with the games i tried everything seems to be installing and syncing.

Please let me know how it works out so that way i can add in this fix to the Decky Plugin Version(NSLPlugin.desktop)

**Comment by moraroy** [comment]
also if it asks to update DO NOT UPDATE haha it will update you back to 2.0.80

**Comment by Ryonez** [comment]
I find some amusement after watching people talking about changing the proton version, that the fix implemented is what I suggested back at the start, not using the beta build, and do not update.  >.<

It's been of some concern to me, given the planned removal of seperate app id's for the stores. A feature I quite like and currently use. But if it's removed, it seems like most here would've downgraded proton, which would mean everything would've been using an out of date one.

Glad to see it worked on though, thank you. Wonder if there's a way to tell the client to ignore beta builds for users, most would be inclined to update when the client offers it.

**Comment by moraroy** [comment]
@Ryonez /runWithoutUpdating

**Comment by CristianJaviM** [comment]
I guess the new new version of gog broke this again. I don't have my Legion Go S at hand, but last night I tried installing through NSL and the storefronts/launchers themselves worked. But trying to install a game, it shows the E2 error again.

It seems that another beta version sneaked and broke things.

**Comment by R4Zi3L** [comment]
Same with GOG 2.1.8 but E4 (it updated automaticly itself)
It will download a few MB and disconnect or wont even connect to download another game
